game-core/src/main/java/games/strategy/triplea/attachments/UnitSupportAttachment.java

```diff
@@ -214,7 +215,19 @@ private void resetNumber() {
     number = 0;
   }
 
-  private void setBonusType(final String type) {
+  private void setBonusType(final String type) throws GameParseException {
+    final String[] s = splitOnColon(type);
```

Comment:
Just occurred to me, with a value object, we can have a constructor encapsulate this logic entirely, eg:

```
bonusType = new BonusType(type);
```
That opens up the parsing for easy testing and encapsulates the construction logic.